src/query/ast/src/parser/token.rs

```diff
@@ -1094,3 +1096,11 @@ impl TokenKind {
         }
     }
 }
+
+pub fn all_reserved_keywords() -> Vec<String> {
+    let mut result = Vec::new();
+    for token in TokenKind::iter() {
```

Comment:
Not all keywords are reserved. You can query it via `token.is_reserved_ident(after_as: bool)` and `token.is_reserved_function_name(after_as: bool)`. `after_as = true` will make some of the reserved keywords unreserved because such keywords are unambiguous after an `AS`.